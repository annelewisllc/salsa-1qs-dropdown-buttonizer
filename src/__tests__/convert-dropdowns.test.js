const fs = require('fs');
const { convertDropdownsToRadios } = require('../convert-dropdowns');

const htmlMock = fs.readFileSync(`${__dirname}/../__mocks__/salsa-action.html`);

describe('convertDropdownsToRadios', () => {
  beforeEach(() => {
    document.body.innerHTML = htmlMock;
  });

  it('creates a OQS div that matches the expected snapshot', () => {
    expect.assertions(2);
    convertDropdownsToRadios();
    expect(document.body.querySelectorAll('.oqs')).not.toBeNull();
    document.body.querySelectorAll('.oqs').forEach(oqs => {
      expect(oqs).toMatchSnapshot();
    })
  });

  it('removes the original select element', () => {
    expect.assertions(1);
    convertDropdownsToRadios();
    expect(document.body.querySelector('select[name="do_you_trust_donald_trump_"]')).toBeNull();
  });
});
