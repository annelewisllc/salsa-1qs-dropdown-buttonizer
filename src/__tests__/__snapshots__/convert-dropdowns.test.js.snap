// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertDropdowns creates a OQS div that matches the expected snapshot 1`] = `
<div
  class="oqs"
>
  
              
  <input
    checked=""
    id="do_you_trust_donald_trump__No_id"
    name="do_you_trust_donald_trump_"
    type="radio"
    value="No"
  />
  
              
  <label
    class="highlight"
    for="do_you_trust_donald_trump__No_id"
  >
    No
  </label>
  
            
              
  <input
    id="do_you_trust_donald_trump__Yes_id"
    name="do_you_trust_donald_trump_"
    type="radio"
    value="Yes"
  />
  
              
  <label
    for="do_you_trust_donald_trump__Yes_id"
  >
    Yes
  </label>
  
            
</div>
`;
